application/src/main/java/org/togetherjava/tjbot/features/utils/LinkDetection.java

```diff
@@ -0,0 +1,83 @@
+package org.togetherjava.tjbot.features.utils;
+
+import com.linkedin.urls.Url;
+import com.linkedin.urls.detection.UrlDetector;
+import com.linkedin.urls.detection.UrlDetectorOptions;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Utility class to detect links.
+ */
+public class LinkDetection {
+
+    /**
+     * Possible ways to filter a link.
+     *
+     * @see LinkDetection
+     */
+    public enum LinkFilter {
+        /**
+         * Filters links suppressed with {@literal <url>}.
+         */
+        SUPPRESSED,
+        /**
+         * Filters links that are not using http scheme.
+         */
+        NON_HTTP_SCHEME
+    }
+
+    private LinkDetection() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    /**
+     * Extracts all links from the given content.
+     *
+     * @param content the content to search through
+     * @param filter the filters applied to the urls
+     * @return a list of all found links, can be empty
+     */
+    public static List<String> extractLinks(String content, Set<LinkFilter> filter) {
+        return new UrlDetector(content, UrlDetectorOptions.BRACKET_MATCH).detect()
+            .stream()
+            .map(url -> toLink(url, filter))
+            .flatMap(Optional::stream)
+            .toList();
+    }
+
+    /**
+     * Checks whether the given content contains a link.
+     *
+     * @param content the content to search through
+     * @return true if the content contains at least one link
+     */
+    public static boolean containsLink(String content) {
+        return !(new UrlDetector(content, UrlDetectorOptions.BRACKET_MATCH).detect().isEmpty());
```

Comment:
The parenthesis is useless here.

Reply:
how is it useless? its needed for `!` I think